statsmodels/tsa/statespace/tbats.py

```diff
@@ -0,0 +1,1436 @@
+                 reflect new params
+        """
+        # We need to handle positional arguments in two ways, in case this was
+        # called by a Scipy optimization routine
+        if len(args) > 0:
+            argnames = ['transformed', 'complex_step']
+            # the fit() method will pass a dictionary
+            if isinstance(args[0], dict):
+                flags = args[0]
+            # otherwise, a user may have just used positional arguments...
+            else:
+                flags = dict(zip(argnames, args))
+            transformed = flags.get('transformed', True)
+            complex_step = flags.get('complex_step', True)
+
+            for name, value in flags.items():
+                if name in kwargs:
+                    raise TypeError("loglike() got multiple values for keyword"
+                                    " argument '%s'" % name)
+        else:
+            transformed = kwargs.pop('transformed', True)
+            complex_step = kwargs.pop('complex_step', True)
+
+        if not transformed:
+            params = self.transform_params(params)
+
+        self.update(params, transformed=True, complex_step=complex_step)
+
+        if complex_step:
+            kwargs['inversion_method'] = INVERT_UNIVARIATE | SOLVE_LU
+
+        sse = self.ssm.loglike(complex_step=complex_step, **kwargs)
+
+        lmbda = params[1]
+        if self.boxcox:
+            sse += (lmbda - 1) * np.nansum(self.data.log_endog[self.loglikelihood_burn:])
+
+        return sse
+
+    def loglikeobs(self, params, transformed=True, complex_step=False,
+                   **kwargs):
+        params = np.asarray(params, dtype=np.float64)
+        if transformed is False:
+            params = self.transform_params(params)
+            transformed = True
+
+        kwargs.setdefault('loglikelihood_burn', self.loglikelihood_burn)
+
+        loglikelihood_burn = kwargs.get('loglikelihood_burn')
+
+        loglikeobs = super().loglikeobs(params, transformed, complex_step, **kwargs)
+
+        if self.boxcox:
+            lmbda = params[1]
+            loglikeobs += (lmbda - 1) * self.data.log_endog
+
+        loglikeobs[:loglikelihood_burn] = 0
+
+        return loglikeobs
+
+    def fit(self, *args, **kwargs):
+        kwargs.setdefault('maxiter', (self.k_params * 200) ** 2)
+        return super().fit(*args, **kwargs)
+
+    @property
+    def start_params(self):
+
+        if not hasattr(self, 'parameters'):
+            return []
+
+        _start_params = {}
+
+        endog = self.endog
+        exog = self.exog
+
+        _start_params['obs_var'] = .001
+
+        if np.any(np.isnan(endog)):
+            endog = endog[~np.isnan(endog)]
+            if exog is not None:
+                exog = exog[~np.isnan(endog)]
+
+        if self.boxcox:
+            endog, lmbda = self.transform_boxcox(self.data.endog)
+            if np.isnan(lmbda) or not (0 <= lmbda <= 1):
```

Comment:
As of the next commit you can specify `bounds=(0, 1)` as an optional argument to `transform_boxcox`. This will ensure optimisation takes place over the specified interval.